akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala

```diff
@@ -35,6 +33,20 @@ import akka.stream.stage._
 object MergeHub {
   private val Cancel = -1
 
+  /**
+   * A DrainingControl object is created during the materialization of a MergeHub and allows to initiate the draining
+   * and eventual completion of the Hub from the outside.
+   */
+  sealed trait DrainingControl {
```

Comment:
Maybe it would make sense to call this trait `MergeHubControl` and also include the sink in it, so we can later add more functionality more easily without having to touch the API again.

Reply:
Indeed I tried to align the naming with Alpakka Kafka. As a user I think is good to have the API have similar names and behaviour as it lowers the cognitive burden. 

Regarding moving the `Sink` I second @octonato, I fear that it will make the API a little less obvious, again increasing the cognitive burden on the users.

Reply:
I agree that keeping it aligned is good, so lets keep the name. I just noticed this one is missing the "not for user extension part" though. That allows us to add methods or change the hierarchy later, potentially introducing some other trait if needed. So please add that.